### PR TITLE
feat: [PM4-55] refactored table component. search and reservationlist connected

### PR DESCRIPTION
**Pull request by SandroGuerotto**
search and my-reservation pages are connected to backend.
This can be toggled with useApi.
Additionally, changed the table-component to handle JSX.Element in preparation for PM4-67.
Added a default buffer to use for the fetch requests

### CODE REVIEW

File: frontend/src/components/loading-buffer/loading-buffer.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { CircularProgress } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+
+export const Loading = ({ loading }) => {
```
Comment (DaRabus) [inline frontend/src/components/loading-buffer/loading-buffer.tsx:5]: Not sure why you pass the state with it, you can just call the component when it's "loading" and then disable it again

File: frontend/src/components/table/table-component.tsx [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
+import CustomTableRow, { RowDataType } from './table-row';
 import Paper from '@mui/material/Paper';
 import TableHeader from './table-header';
 
 interface TableComponentProps {
   headerNames: string[];
   onRowClick: (row: any) => void;
-  onCellClick: Map<string, (cell: any) => void>;
-  data: Array<string[]>;
+  data: Array<RowDataType>;
```
Comment (DaRabus) [inline frontend/src/components/table/table-component.tsx:12]:
```suggestion
  data: RowDataType[];
```

**Review by DaRabus — APPROVED** [review]
Looks good, some minor stuff to change rest looks ok, and is mostly taste thingy

File: frontend/src/components/loading-buffer/loading-buffer.tsx (hunk 1 above)
Comment (SandroGuerotto) [reply frontend/src/components/loading-buffer/loading-buffer.tsx:5]: I will fix it later when more components call the backend

**Pull request merged**